theano/tensor/basic.py

```diff
@@ -6093,3 +6093,12 @@ def c_code_cache_version(self):
 
     def do_constant_folding(self, node):
         return False
+
+    def connection_pattern(self, node):
+        return [[False] for i in node.inputs]
+
+    def grad(self, inputs, grads):
+        return [DisconnectedType()() for i in inputs]
+
+    def R_op(self, inputs, eval_points):
+        return [zeros(inputs, self.dtype)]
```

Comment:
@carriepl I don't think that this restriction exists, or if it does in theory, then it has never been a problem that [`Alloc.R_op`](https://github.com/carriepl/Theano/blob/scan_speedup_cgt/theano/tensor/basic.py#L2805-L2808) does not follow it.
Btw, looking at `Alloc.R_op`, I would be tempted to just return `[None]` for `AllocEmpty.R_op`, since `AllocEmpty` has no differentiable input, at least if it works. If it does not work, then we should probably fix `Alloc.R_op` to return zeros instead of None in the first case.